nupic/research/frameworks/pytorch/imagenet/imagenet_tune.py

```diff
@@ -338,16 +338,29 @@ def _process_config(self, config):
         self.sigopt = None
         if "sigopt_config" in config:
             assert config.get("sigopt_experiment_id", None) is not None
-            self.sigopt = SigOptImagenetExperiment(
+
+            # Check for user specified sigopt-experiment class.
+            experiment_class = config.get(
+                "sigopt_experiment_class", "SigOptImagenetExperiment")
+            experiment_class = getattr(sigopt_experiment, experiment_class)
+
+            # Instantiate experiment.
+            self.sigopt = experiment_class(
                 experiment_id=config["sigopt_experiment_id"],
                 sigopt_config=config["sigopt_config"])
+
+            # Get suggestion and update config.
             self.suggestion = self.sigopt.get_next_suggestion()
             self.sigopt.update_config_with_suggestion(config, self.suggestion)
             print("SigOpt suggestion: ", self.suggestion)
             print("Config after Sigopt:")
             pprint(config)
             self.epochs = config["epochs"]
 
+            # Get names of performance metrics.
+            assert "metrics" in config["sigopt_config"]
+            self.metric_names = [metric["name"] for metric in config["sigopt_config"]["metrics"]]
```

Comment:
Should we assert that at least one of the metrics is called 'mean_accuracy'?

Reply:
Not necessarily. It's hard to think of a case where one wouldn't use 'mean_accuracy', ut the user has no requirement to do so

Reply:
But if they don't use mean_accuracy, the if statement in _process_result() would prevent the observation from being recorded:

`if result["mean_accuracy"] > 0.0:`

Reply:
Good point. Didn't think of that. See
https://github.com/numenta/nupic.research/pull/217